Eatery Watch App Extension/UI/Eateries/CampusEateriesView.swift

```diff
@@ -0,0 +1,201 @@
+                    self.viewData.fetchLocation { error in
+                        if self.sortMethod == .distance {
+                            self.sortMethod = .name
+                            self.errorInfo = ErrorInfo(title: "Could not get location.", error: error)
+                        }
+                    }
+                }
+            }
+            .animation(nil)
+
+            AreaFilterView(self.$areaFilter).animation(nil)
+
+            HStack {
+                Text("Open").font(.headline)
+                Spacer()
+            }
+
+            if !open.isEmpty {
+                self.eateriesView(open)
+            } else {
+                Group {
+                    Text("No Open Eateries")
+                }
+            }
+
+            HStack {
+                Text("Closed").font(.headline)
+                Spacer()
+            }
+
+            if !closed.isEmpty {
+                self.eateriesView(closed)
+            } else {
+                Group {
+                    Text("No Closed Eateries")
+                }
+            }
+        }
+        .navigationBarTitle("Eateries")
+        .animation(.easeOut(duration: 0.25))
+        .contextMenu {
+            Button(action: {
+                self.viewData.fetchCampusEateries { error in
+                    self.errorInfo = ErrorInfo(title: "Could not fetch eateries.", error: error)
+                }
+            }, label: {
+                VStack{
+                    Image(systemName: "arrow.clockwise")
+                        .font(.title)
+                    Text("Refresh Eateries")
+                }
+            })
+        }
+        .alert(item: self.$errorInfo) { errorInfo -> Alert in
+            Alert(title: Text("Error: ") + Text(errorInfo.title),
+                  message: Text(errorInfo.error.localizedDescription),
+                  dismissButton: .default(Text("OK")))
+        }
+        .onAppear {
+            if self.firstAppearance {
+                self.viewData.fetchCampusEateries { error in
+                    self.errorInfo = ErrorInfo(title: "Could not fetch eateries.", error: error)
+                }
+
+                self.firstAppearance = false
+            }
+        }
+    }
+
+    func eateriesView(_ eateries: [CampusEatery]) -> some View {
+        ForEach(eateries) { eatery in
+            NavigationLink(destination: CampusEateryView(eatery: eatery)) {
+                CampusEateryRow(eatery: eatery, userLocation: self.sortMethod == .distance ? self.viewData.userLocation : nil)
+            }
+        }
+    }
+
+    private func filter(_ eateries: [CampusEatery]) -> [CampusEatery] {
+        if let areaFilter = self.areaFilter {
+            return eateries.filter { eatery in
+                eatery.area == areaFilter
+            }
+        } else {
+            return eateries
+        }
+    }
+
+    private func sort(_ eateries: [CampusEatery]) -> [CampusEatery] {
```

Comment:
can we rename this to `sorted`. i believe `sort` is typically implied to sort the array in place without returning any value